rules/legacy_xcodeproj.bzl

```diff
@@ -881,6 +897,18 @@ def _populate_xcodeproj_targets_and_schemes(ctx, targets, src_dot_dots, all_tran
             target_settings["TARGETED_DEVICE_FAMILY"] = target_info.targeted_device_family
 
         pre_build_scripts = []
+
+        if virtualize_frameworks:
+            pre_build_scripts.append({
```

Comment:
Is there a way that we could prevent syncing frameworks when `virtualize_frameworks` is set - perhaps pass a new argument to the installed script.

If a developer builds a framework AFAIK there is an issue with it

Reply:
Most issues I've found were when building a framework, building apps taking dep on fmws seems to be bypassing some edge cases here.

The malformed frameworks being created under `DerivedData` are created when Xcode launches by `XCBBuildService` and not our post-build scripts (elaborated more [here](https://github.com/bazel-ios/rules_ios/pull/661#discussion_r1122080506)). Since the current generator is missing a lot of logic to get this right I believe a pre-build cleanup is a more effective workaround. Def not ideal but again, we're moving to `XCHammer` soon.